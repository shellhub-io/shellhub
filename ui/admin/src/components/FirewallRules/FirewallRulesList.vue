<template>
  <DataTable
    :headers
    :items="firewallRules"
    v-model:itemsPerPage="itemsPerPage"
    v-model:page="page"
    :loading
    :totalCount="firewallRulesCount"
    :itemsPerPageOptions="[10, 20, 50, 100]"
    data-test="firewall-rules-list"
  >
    <template v-slot:rows>
      <tr v-for="(firewallRule, index) in firewallRules" :key="index">
        <td>
          {{ firewallRule.tenant_id }}
        </td>
        <td>
          {{ firewallRule.priority }}
        </td>
        <td class="text-capitalize">
          {{ firewallRule.action }}
        </td>
        <td>
          {{ formatSourceIP(firewallRule.source_ip) }}
        </td>
        <td>
          {{ formatUsername(firewallRule.username) }}
        </td>
        <td>
          <div v-if="isHostname(firewallRule.filter)">
            {{ formatHostnameFilter(firewallRule.filter) }}
          </div>
          <div v-else>
            <v-tooltip
              v-for="(tag, index) in firewallRule.filter.tags"
              :key="index"
              bottom
              :disabled="!showTag(tag.name)"
            >
              <template #activator="{ props }">
                <v-chip
                  class="mr-1"
                  density="compact"
                  outlined
                  v-bind="props"
                >
                  {{ displayOnlyTenCharacters(tag.name) }}
                </v-chip>
              </template>

              <span>
                {{ tag.name }}
              </span>
            </v-tooltip>
          </div>
        </td>
        <td>
          <v-tooltip bottom anchor="bottom">
            <template v-slot:activator="{ props }">
              <v-icon
                tag="a"
                dark
                v-bind="props"
                @click="goToFirewallRule(firewallRule.id)"
                @keypress.enter="goToFirewallRule(firewallRule.id)"
                tabindex="0"
                icon="mdi-information"
              />
            </template>
            <span>Details</span>
          </v-tooltip>
        </td>
      </tr>
    </template>
  </DataTable>
</template>

<script setup lang="ts">
import { computed, onMounted, ref, watch } from "vue";
import { useRouter } from "vue-router";
import useFirewallRulesStore from "@admin/store/modules/firewall_rules";
import isHostname from "@/utils/isHostname";
import useSnackbar from "@/helpers/snackbar";
import DataTable from "@/components/Tables/DataTable.vue";
import showTag from "@/utils/tag";
import { displayOnlyTenCharacters, formatHostnameFilter, formatSourceIP, formatUsername } from "@/utils/string";
import handleError from "@/utils/handleError";

const router = useRouter();
const snackbar = useSnackbar();
const firewallRulesStore = useFirewallRulesStore();
const firewallRules = computed(() => firewallRulesStore.firewallRules);
const firewallRulesCount = computed(() => firewallRulesStore.firewallRulesCount);
const loading = ref(false);
const page = ref(1);
const itemsPerPage = ref(10);
const headers = ref([
  {
    text: "Tenant Id",
    value: "tenant_id",
  },
  {
    text: "Priority",
    value: "priority",
  },
  {
    text: "Action",
    value: "action",
  },
  {
    text: "Source Ip",
    value: "source_ip",
  },
  {
    text: "Username",
    value: "username",
  },
  {
    text: "Filter",
    value: "filter",
  },
  {
    text: "Actions",
    value: "actions",
  },
]);

const goToFirewallRule = (ruleId: string) => router.push({ name: "firewallRulesDetails", params: { id: ruleId } });

const fetchFirewallRules = async () => {
  try {
    loading.value = true;
    await firewallRulesStore.fetchFirewallRulesList({
      page: page.value,
      perPage: itemsPerPage.value,
    });
  } catch (error) {
    handleError(error);
    snackbar.showError("Failed to fetch firewall rules.");
  }
  loading.value = false;
};

watch([itemsPerPage, page], async () => {
  await fetchFirewallRules();
});

onMounted(async () => {
  await fetchFirewallRules();
});

defineExpose({
  headers,
  loading,
  itemsPerPage,
  page,
  firewallRules,
  formatSourceIP,
  formatUsername,
  formatHostnameFilter,
  displayOnlyTenCharacters,
  showTag,
});
</script>
