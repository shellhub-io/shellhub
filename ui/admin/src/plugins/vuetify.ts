// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
