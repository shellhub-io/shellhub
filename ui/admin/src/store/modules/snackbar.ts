import { Module } from "vuex";
import { State } from "./../index";

export interface snackbarMessageAndContentType {
  typeMessage: string;
  typeContent: string;
}

export interface SnackbarState {
  snackbarError: boolean;
  snackbarSuccess: boolean;
  snackbarMessageAndContentType: snackbarMessageAndContentType;
  snackbarCopy: boolean;
}

export const snackbar: Module<SnackbarState, State> = {
  namespaced: true,

  state: {
    snackbarError: false,
    snackbarSuccess: false,
    snackbarMessageAndContentType: { typeMessage: "", typeContent: "" },
    snackbarCopy: false,
  },

  getters: {
    snackbarSuccess: (state) => state.snackbarSuccess,
    snackbarError: (state) => state.snackbarError,
    snackbarMessageAndContentType: (state) => state.snackbarMessageAndContentType,
    snackbarCopy: (state) => state.snackbarCopy,
  },

  mutations: {
    setSnackbarSuccessAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarNoContent: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarSuccessDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarSuccess = true;
    },

    unsetSnackbarSuccess: (state) => {
      state.snackbarSuccess = false;
    },

    setSnackbarErrorLoadingOrAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorCustom: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorDefault: (state) => {
      state.snackbarMessageAndContentType = { typeMessage: "default", typeContent: "" };
      state.snackbarError = true;
    },

    setSnackbarErrorLicense: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    unsetSnackbarError: (state) => {
      state.snackbarError = false;
    },

    setSnackbarCopy: (state, value) => {
      state.snackbarMessageAndContentType = { typeMessage: "", typeContent: value };
      state.snackbarCopy = true;
    },

    unsetSnackbarCopy: (state) => {
      state.snackbarCopy = false;
    },
  },

  actions: {
    showSnackbarSuccessAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarSuccessAction", data);
    },

    showSnackbarSuccessDefault: ({ commit }) => {
      commit("setSnackbarSuccessDefault");
    },

    unsetShowStatusSnackbarSuccess: ({ commit }) => {
      commit("unsetSnackbarSuccess");
    },

    showSnackbarErrorLoading: ({ commit }, value) => {
      const data = { typeMessage: "loading", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarNoContent: ({ commit }) => {
      const data = { typeMessage: "no-content", typeContent: "" };
      commit("setSnackbarNoContent", data);
    },

    showSnackbarErrorAction: ({ commit }, value) => {
      const data = { typeMessage: "action", typeContent: value };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorCustom: ({ commit }, value) => {
      const data = { typeMessage: "custom", typeContent: value };
      commit("setSnackbarErrorCustom", data);
    },

    showSnackbarErrorLicense: ({ commit }, value) => {
      const data = { typeMessage: "licenseRequired", typeContent: value };
      commit("setSnackbarErrorLicense", data);
    },

    showSnackbarErrorDefault: ({ commit }) => {
      commit("setSnackbarErrorDefault");
    },

    unsetShowStatusSnackbarError: ({ commit }) => {
      commit("unsetSnackbarError");
    },

    showSnackbarCopy: ({ commit }, value) => {
      commit("setSnackbarCopy", value);
    },

    unsetShowStatusSnackbarCopy: ({ commit }) => {
      commit("unsetSnackbarCopy");
    },
  },
};
