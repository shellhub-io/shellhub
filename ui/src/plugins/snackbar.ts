import { App, Plugin, reactive } from "vue";

const SnackbarInjectionKey = Symbol("snackbar");

type SnackbarType = "success" | "error" | "info" | "warning";

interface SnackbarState {
  message: string;
  type: SnackbarType;
  show: boolean;
}

interface ISnackbarPlugin {
  showSuccess: (msg: string) => void;
  showError: (msg: string) => void;
  showInfo: (msg: string) => void;
  showWarning: (msg: string) => void;
  getMessage: () => string;
  getType: () => SnackbarType;
  getShow: () => boolean;
}

const state = reactive<SnackbarState>({
  message: "",
  type: "info",
  show: false,
});

const showSnackbar = (type: SnackbarType, message: string) => {
  state.message = message;
  state.type = type;
  state.show = true;

  setTimeout(() => {
    state.show = false;
  }, 4000);
};

const plugin: ISnackbarPlugin = {
  showSuccess: (msg: string) => showSnackbar("success", msg),
  showError: (msg: string) => showSnackbar("error", msg),
  showInfo: (msg: string) => showSnackbar("info", msg),
  showWarning: (msg: string) => showSnackbar("warning", msg),
  getMessage: () => state.message,
  getType: () => state.type,
  getShow: () => state.show,
};

const SnackbarPlugin: Plugin = {
  install(app: App) {
    app.provide(SnackbarInjectionKey, plugin);
  },
};

export { SnackbarPlugin, SnackbarInjectionKey, plugin };
export type { SnackbarState, ISnackbarPlugin };
