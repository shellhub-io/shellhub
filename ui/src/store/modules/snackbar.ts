import { Module } from "vuex";
import { State } from "..";

export interface snackbarMessageAndContentType {
  typeMessage: string;
  typeContent: string;
}

export interface SnackbarState {
  message: string,
  type: string,
  snackbarError: boolean;
  snackbarSuccess: boolean;
  snackbarMessageAndContentType: snackbarMessageAndContentType;
  snackbarCopy: boolean;
}

export const snackbar: Module<SnackbarState, State> = {
  namespaced: true,
  state: {
    message: "",
    type: "",
    snackbarError: false,
    snackbarSuccess: false,
    snackbarMessageAndContentType: {
      typeMessage: "",
      typeContent: "",
    },
    snackbarCopy: false,
  },

  getters: {
    snackbarSuccess: (state) => state.snackbarSuccess,
    snackbarError: (state) => state.snackbarError,
    snackbarMessageAndContentType: (state) => state.snackbarMessageAndContentType,
    snackbarCopy: (state) => state.snackbarCopy,
  },

  mutations: {
    showMessage: (state, data) => {
      state.type = data.type;
      state.message = data.message;
    },

    setSnackbarSuccessAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    unsetSnackbarSuccess: (state) => {
      state.snackbarSuccess = false;
    },

    setSnackbarSuccessNotRequest: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarSuccess = true;
    },

    setSnackbarErrorLoadingOrAction: (state, data) => {
      state.snackbarMessageAndContentType = data;
      state.snackbarError = true;
    },

    setSnackbarErrorDefault: (state) => {
      state.snackbarMessageAndContentType = {
        typeMessage: "default",
        typeContent: "",
      };
      state.snackbarError = true;
    },

    unsetSnackbarError: (state) => {
      state.snackbarError = false;
    },

    setSnackbarCopy: (state, value) => {
      state.snackbarMessageAndContentType = {
        typeMessage: "",
        typeContent: value,
      };
      state.snackbarCopy = true;
    },

    unsetSnackbarCopy: (state) => {
      state.snackbarCopy = false;
    },
  },

  actions: {
    showSnackbarSuccessAction: ({ commit }, value) => {
      const data = {
        typeMessage: "action",
        typeContent: value,
      };
      commit("setSnackbarSuccessAction", data);
    },

    unsetShowStatusSnackbarSuccess: ({ commit }) => {
      commit("unsetSnackbarSuccess");
    },

    showSnackbarSuccessNotRequest: (context, value) => {
      const data = {
        typeMessage: "notRequest",
        typeContent: value,
      };
      context.commit("setSnackbarSuccessNotRequest", data);
    },

    showSnackbarErrorLoading: ({ commit }, value) => {
      const data = {
        typeMessage: "loading",
        typeContent: value,
      };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorAction: ({ commit }, value) => {
      const data = {
        typeMessage: "action",
        typeContent: value,
      };
      commit("setSnackbarErrorLoadingOrAction", data);
    },

    showSnackbarErrorDefault: ({ commit }) => {
      commit("setSnackbarErrorDefault");
    },

    unsetShowStatusSnackbarError: ({ commit }) => {
      commit("unsetSnackbarError");
    },

    showSnackbarCopy: ({ commit }, value) => {
      commit("setSnackbarCopy", value);
    },

    unsetShowStatusSnackbarCopy: ({ commit }) => {
      commit("unsetSnackbarCopy");
    },
  },
};
