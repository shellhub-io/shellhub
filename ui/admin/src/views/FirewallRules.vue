<template>
  <div>
    <h1>Firewall Rules</h1>
  </div>
  <v-card class="mt-2">
    <FirewallRulesList />
  </v-card>
</template>

<script lang="ts">
import { defineComponent } from "vue";
import FirewallRulesList from "../components/FirewallRules/FirewallRulesList.vue";

export default defineComponent({
  components: { FirewallRulesList },
});
</script>
