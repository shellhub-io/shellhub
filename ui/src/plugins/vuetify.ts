// Styles
import "@mdi/font/css/materialdesignicons.css";
import "@fortawesome/fontawesome-free/css/all.css";
import "vuetify/styles";
import { VFileUpload, VFileUploadItem } from "vuetify/labs/VFileUpload";
import "../styles/_variables.scss";
import { aliases as faAliases, fa } from "vuetify/iconsets/fa";

// Vuetify
import { createVuetify } from "vuetify";

const light = {
  dark: false,
  colors: {
    primary: "#667acc",
    secondary: "#FFFFFF",
    background: "#F5F5F5",
    tabs: "#F5F5F5",
    foreground: "#F5F5F5",
    paymentForm: "#F5F5F5",
    terminal: "#0f1526",
    "v-theme-background": "#FFFFFF",
    "v-theme-surface": "#FFFFFF",
    "v-theme-card": "#FFFFFF",
  },
};

const dark = {
  dark: true,
  colors: {
    primary: "#667acc",
    secondary: "#1E2127",
    background: "#18191B",
    tabs: "#1E1E1E",
    foreground: "#1E1E1E",
    paymentForm: "#E0E0E0",
    terminal: "#0f1526",
    "v-theme-background": "#1E2127",
    "v-theme-surface": "#1E2127",
    "v-theme-card": "#22252B",
  },
};

export default createVuetify({
  components: {
    VFileUpload,
    VFileUploadItem,
  },
  theme: {
    defaultTheme: "dark",
    themes: {
      dark,
      light,
    },
  },
  defaults: {
    global: {

    },
    VTextField: {
      variant: "outlined",
      density: "comfortable",
    },
    VSelect: {
      variant: "outlined",
      density: "comfortable",
    },
    VTextarea: {
      variant: "outlined",
    },
  },
  icons: {
    faAliases,
    sets: {
      fa,
    },
  },
});
