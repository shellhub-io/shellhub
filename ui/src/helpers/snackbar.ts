import { inject } from "vue";
import { InjectionKey, type ISnackbarPlugin } from "@/plugins/snackbar";

const useSnackbar = (): ISnackbarPlugin => inject(InjectionKey) as ISnackbarPlugin;

export default useSnackbar;
