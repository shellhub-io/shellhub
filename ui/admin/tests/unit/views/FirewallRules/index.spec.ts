import { createVuetify } from "vuetify";
import { mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import { createPinia, setActivePinia } from "pinia";
import useFirewallRulesStore from "@admin/store/modules/firewall_rules";
import { SnackbarPlugin } from "@/plugins/snackbar";
import routes from "../../../../src/router";
import FirewallRules from "../../../../src/views/FirewallRules.vue";

type FirewallRulesWrapper = VueWrapper<InstanceType<typeof FirewallRules>>;

describe("Firewall Rules", () => {
  let wrapper: FirewallRulesWrapper;

  beforeEach(() => {
    const pinia = createPinia();
    setActivePinia(pinia);

    const firewallStore = useFirewallRulesStore();
    vi.spyOn(firewallStore, "getNumberFirewalls", "get").mockReturnValue(1);
    firewallStore.fetch = vi.fn();

    const snackbarStore = useSnackbarStore();
    snackbarStore.showSnackbarErrorAction = vi.fn();

    const vuetify = createVuetify();

    wrapper = mount(FirewallRules, {
      global: {
        plugins: [pinia, vuetify, routes, SnackbarPlugin],
      },
    });
  });

  it("Is a Vue instance", () => {
    expect(wrapper).toBeTruthy();
  });

  it("Renders the component", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Should render all the components in the screen", () => {
    expect(wrapper.find("h1").text()).toContain("Firewall Rules");
    expect(wrapper.find("[data-test='firewallRules-list']").exists()).toBe(true);
  });
});
