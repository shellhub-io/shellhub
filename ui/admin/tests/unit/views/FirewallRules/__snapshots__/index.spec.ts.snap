// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Firewall Rules > Renders the component 1`] = `
"<h1 class="mb-2">Firewall Rules</h1>
<div data-test="firewall-rules-list">
  <div class="v-table v-theme--light v-table--density-default bg-background border rounded text-center" data-test="datatable">
    <!---->
    <div class="v-table__wrapper">
      <table>
        <thead class="bg-v-theme-background">
          <tr>
            <th class="text-center" data-test="th-tenant_id"><span data-test="th-label">Tenant Id</span></th>
            <th class="text-center" data-test="th-priority"><span data-test="th-label">Priority</span></th>
            <th class="text-center" data-test="th-action"><span data-test="th-label">Action</span></th>
            <th class="text-center" data-test="th-source_ip"><span data-test="th-label">Source Ip</span></th>
            <th class="text-center" data-test="th-username"><span data-test="th-label">Username</span></th>
            <th class="text-center" data-test="th-filter"><span data-test="th-label">Filter</span></th>
            <th class="text-center" data-test="th-actions"><span data-test="th-label">Actions</span></th>
          </tr>
        </thead>
        <tbody data-test="tbody-has-items">
          <tr>
            <td>xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx</td>
            <td>4</td>
            <td class="text-capitalize">allow</td>
            <td>127.0.0.1</td>
            <td>shellhub</td>
            <td>
              <div><span class="v-chip v-theme--light v-chip--density-compact v-chip--size-default v-chip--variant-tonal mr-1" draggable="false" outlined="" aria-describedby="v-tooltip-v-7"><!----><span class="v-chip__underlay"></span>
                <!---->
                <!---->
                <div class="v-chip__content" data-no-activator="">test-tag</div>
                <!---->
                <!----></span>
                <!--teleport start-->
                <!--teleport end-->
              </div>
            </td>
            <td><a class="mdi-information mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable" role="button" aria-hidden="false" tabindex="0" dark="" aria-describedby="v-tooltip-v-10"></a>
              <!--teleport start-->
              <!--teleport end-->
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <!---->
  </div>
  <!--v-if-->
  <div class="d-flex w-100 justify-end align-center" data-test="pager"><span class="text-subtitle-2 mr-4" data-test="ipp-label">Items per page:</span>
    <div>
      <div class="v-input v-input--horizontal v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-input--plain-underlined v-combobox v-combobox--single mb-4" data-test="ipp-combo">
        <!---->
        <div class="v-input__control">
          <div class="v-field v-field--active v-field--appended v-field--dirty v-field--no-label v-field--variant-underlined v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="menu" aria-expanded="false" aria-controls="menu-v-0">
            <div class="v-field__overlay"></div>
            <div class="v-field__loader">
              <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                <!---->
                <div class="v-progress-linear__background"></div>
                <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <div class="v-progress-linear__indeterminate">
                    <div class="v-progress-linear__indeterminate long"></div>
                    <div class="v-progress-linear__indeterminate short"></div>
                  </div>
                </transition-stub>
                <!---->
              </div>
            </div>
            <!---->
            <div class="v-field__field" data-no-activator="">
              <!---->
              <!---->
              <!---->
              <div class="v-field__input" data-no-activator="">
                <!---->
                <!---->
                <div class="v-combobox__selection"><span class="v-combobox__selection-text">10<!----></span></div><input size="1" role="combobox" type="text" id="input-v-2" aria-expanded="false" aria-controls="menu-v-0" outlined="" value="10">
              </div>
              <!---->
            </div>
            <!---->
            <div class="v-field__append-inner">
              <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-icon--clickable v-combobox__menu-icon" role="button" aria-hidden="false" tabindex="-1" aria-label="Open" title="Open"></i>
              <!---->
            </div>
            <div class="v-field__outline">
              <!---->
              <!---->
            </div>
          </div>
        </div>
        <!---->
        <!---->
      </div>
    </div>
    <div class="d-flex align-center" data-test="pager-controls"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled="" data-test="pager-prev"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button><span class="text-subtitle-2" data-test="pager-text">1 of 1</span><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-plain" disabled="" data-test="pager-next"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
        <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
        <!---->
        <!---->
      </button></div>
  </div>
</div>"
`;
